db/migrate/20240402194159_create_version_metadatas.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class CreateVersionMetadatas < ActiveRecord::Migration[7.0]
+  def change
+    create_table :version_metadata, primary_key: %i[druid version] do |t|
+      t.string      :druid, null: false
+      t.integer     :version, default: 1
```

Comment:
should `version` also be `null: false`, or is that implied by it being part of the composite primary key?

i guess if you wanted to avoid the addition of the composite primary key gem (though i'm not personally opposed to it), you could have a typical integer PK, make `druid` and `version` both explicitly non-null, and put a unique index on the columns individually and on the combo of them so you don't have any dupes for the same druid version?

Reply:
I like this idea of getting rid of the gem.  Let me re-do the migration, add a model level validation to ensure we don't have dupes as well as a db constraint and ensure the tests still pass.